tests/adapters/base.py

```diff
@@ -32,23 +40,92 @@ def test_shutdown(self, test_adapter):
     def test_initialize_model_input(self, test_adapter):
         """
         Test that initalizing model input does as intended.
+
+        Extra tests can be adapted depending on what the adapter should actually
+        do when initialised.
         """
         assert not test_adapter._initialized
         test_adapter.initialize_model_input()
         assert test_adapter._initialized
 
+        # TODO: add tests something like below
```

Comment:
What should this test? That the model sets default parameters?

Reply:
maybe nothing, I haven't worked it out yet. Current thought is that we should make this abstract and then the implementer has to decide what to test (if anything)